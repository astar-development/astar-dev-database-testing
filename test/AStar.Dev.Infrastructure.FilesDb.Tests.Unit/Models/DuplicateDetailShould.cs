using AStar.Dev.Infrastructure.FilesDb.Models;
using AStar.Dev.Utilities;
using Shouldly;

namespace AStar.Dev.Infrastructure.FilesDb.Tests.Unit.Models;

public class DuplicateDetailShould
{
    [Fact]
    public void ContainTheExpectedProperties()
        => new DuplicateDetail
           {
               FileHandle        = new ("mock-file-handle"),
               DirectoryName     = new ("mock-directory-name"),
               FileName          = new ("mock-file-name"),
               FileSize          = 1245,
               IsImage           = true,
               FileLastViewed    = new DateTimeOffset(2025, 07, 28, 11, 22, 33, TimeSpan.Zero),
               UpdatedOn         = new (2025, 07, 28, 8, 7, 9, TimeSpan.Zero),
               MoveRequired      = true,
               ImageHeight       = 123,
               ImageWidth        = 345,
               HardDeletePending = new DateTimeOffset(2025, 07, 28, 2, 3, 4, TimeSpan.Zero),
               SoftDeletePending = new DateTimeOffset(2025, 07, 28, 3, 4, 5, TimeSpan.Zero),
               SoftDeleted       = new DateTimeOffset(2025, 07, 28, 5, 6, 1, TimeSpan.Zero),
               Instances         = 2
           }
           .ToJson()
           .ShouldMatchApproved();
}
