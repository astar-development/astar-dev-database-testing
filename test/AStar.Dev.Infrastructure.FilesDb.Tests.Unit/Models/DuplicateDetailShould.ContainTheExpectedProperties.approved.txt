{
  "fileName": "mock-file-name",
  "directoryName": "mock-directory-name",
  "imageHeight": 123,
  "imageWidth": 345,
  "fileSize": 1245,
  "fileHandle": "mock-file-handle",
  "isImage": true,
  "instances": 2,
  "updatedOn": "2025-07-28T08:07:09+00:00",
  "fileLastViewed": "2025-07-28T11:22:33+00:00",
  "softDeleted": "2025-07-28T05:06:01+00:00",
  "softDeletePending": "2025-07-28T03:04:05+00:00",
  "moveRequired": true,
  "hardDeletePending": "2025-07-28T02:03:04+00:00"
}